ios/browser/api/ai_chat/ai_chat.h

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_IOS_BROWSER_API_AI_CHAT_AI_CHAT_H_
+#define BRAVE_IOS_BROWSER_API_AI_CHAT_AI_CHAT_H_
+
+#import <Foundation/Foundation.h>
+#import "ai_chat.mojom.objc.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+OBJC_EXPORT
+@interface AIChat : NSObject
+@property(nonatomic) bool isAgreementAccepted;
+
+@property(nonatomic) bool autoGeneratesQuestions;
+
+@property(nonatomic, readonly) AiChatModel* currentModel;
+
+@property(nonatomic, readonly) NSArray<AiChatModel*>* models;
+
+@property(nonatomic, readonly)
+    NSArray<AiChatConversationTurn*>* conversationHistory;
+
+@property(nonatomic, readonly) bool isRequestInProgress;
+
+@property(nonatomic, readonly) NSArray<NSString*>* suggestedQuestions;
+
+@property(nonatomic, readonly) bool hasPageContent;
+
+@property(nonatomic, readonly) AiChatAPIError currentAPIError;
+
+@property(nonatomic, readonly) bool canShowPremiumPrompt;
+
+- (void)changeModel:(NSString*)modelKey;
+
+- (void)submitHumanConversationEntry:(NSString*)text;
+
+- (void)setConversationActive:(bool)is_conversation_active;
+
+- (void)retryAPIRequest;
+
+- (void)generateQuestions;
+
+- (void)disconnectPageContents;
+
+- (void)clearConversationHistory;
+
+- (void)getPremiumStatus:(void (^_Nullable)(AiChatPremiumStatus))completion;
+
+- (void)rateMessage:(bool)isLiked
+             turnId:(NSUInteger)turnId
+         completion:(void (^_Nullable)(NSString* identifier))completion;
```

Comment:
Missing `_Nullable` on the identifier argument (method impl can call `completion` with `nil`)